lib/connection/connect.js

```diff
@@ -355,4 +354,17 @@ function authenticate(conn, credentials, callback) {
   });
 }
 
+function connectionFailureError(type, err) {
+  switch (type) {
+    case 'error':
+      return new MongoNetworkError(err);
```

Comment:
The reason that I made this `MongoNetworkError(err.message)` in the first place is because the constructor for `MongoError` (which is what the args are eventually passed down to) will convert this _not_ to a `MongoNetworkError`, but instead something that looks like:
```
MongoError {
   name: 'MongoNetworkError'
   ...
}
```

Can you confirm that you haven't undone that here with a test, and imagine a solution if you verify that it is indeed still an issue?

Reply:
It passes the test `should emit `MongoNetworkError` for network errors`, and I've confirmed that it is the error generated at line 360 that is being asserted on.

Reply:
indeed it does! I wonder what issue I was running into..